File: cryptography/src/main/scala/akka/stream/alpakka/cryptography/javadsl/CryptographicFlows.java
/*
 * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
 */

package akka.stream.alpakka.cryptography.javadsl;

import akka.NotUsed;
import akka.stream.alpakka.cryptography.impl.CipherFlow$;
import akka.stream.javadsl.Flow;
import akka.util.ByteString;

import javax.crypto.Cipher;

public class CryptographicFlows {
    public Flow<ByteString, ByteString, NotUsed> cipherFlow(Cipher cipher) {
        return CipherFlow$.MODULE$.cipherFlow(cipher).asJava();
    }
}

File: cryptography/src/main/scala/akka/stream/alpakka/cryptography/scaladsl/CryptographicFlows.scala
/*
 * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
 */

package akka.stream.alpakka.cryptography.scaladsl

import akka.NotUsed
import akka.stream.scaladsl.Flow
import akka.util.ByteString
import akka.stream.alpakka.cryptography.impl.CipherGraphStage

import javax.crypto.Cipher

object CryptographicFlows {
  def cipherFlow(cipher: Cipher): Flow[ByteString, ByteString, NotUsed] =
    Flow.fromGraph(CipherGraphStage(cipher))
}
